sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -661,6 +661,36 @@ class Analyzer(
       // rule: ResolveDeserializer.
       case plan if containsDeserializer(plan.expressions) => plan
 
+      // Some commands (AlterTableDropPartitionCommand) use expressions, so we need to resolve.
+      case c: CommandWithExpression =>
+        val catalogTable = catalog.getTableMetadata(c.getTableName)
+
+        // We should not do this here in order to support general CommandWithExpression
+        if (catalogTable.tableType == CatalogTableType.VIEW) {
+          throw new AnalysisException(
+            "Cannot alter a view with ALTER TABLE. Please use ALTER VIEW instead")
```

Comment:
Except `AlterTableDropPartitionCommand`, do we have other running commands which need this resolving? If not, I think we may not need to add this to `Analyzer` and create new abstraction `CommandWithExpression`.